### DISCUSSION TITLE
custom domain only works for homepage

### DISCUSSION THREAD

**Discussion by yotzu** [community/community#21899] [Other Feature Feedback, Questions, & Ideas]
<p>Im completley stuck on this, I’ve never had this problem before. The issue is I’m using a custom domain on my github pages, the domain works fine for the home page but for any other pages the custom domain doesn’t work. All I get is the <a href="http://github.io" rel="nofollow noopener">github.io</a> sub domain. As in “<a href="http://name.com" rel="nofollow noopener">name.com</a>” works but “<a href="http://name.com/page" rel="nofollow noopener">name.com/page</a>” doesn’t work. Pages will only work with the stock sub domain.</p>
<p>I don’t understand because I’ve never had this issue before. Could it be something I’m doing wrong on github? I’m also using cloudflare, could it be that?</p>
<p>Here is a couple screen shots the first is the correct name. <a href="http://tidyrabbit.com" rel="nofollow noopener">tidyrabbit.com</a> the second should be <a href="http://tidyrabbit.com/test" rel="nofollow noopener">tidyrabbit.com/test</a> but it only works for the github sub domain. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/9cd67d9bf8f8127b733108cdb467d1562c3e2002.png" data-download-href="/uploads/short-url/mns1PrBpeDYbo5sdoARcU9Ifx5w.png?dl=1" title="gh-ss2.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/20893901/181067467-f8056428-14e8-4421-bfbb-317dcd70ea1f.png" alt="gh-ss2.png" data-base62-sha1="mns1PrBpeDYbo5sdoARcU9Ifx5w" width="689" height="269" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/9cd67d9bf8f8127b733108cdb467d1562c3e2002_2_689x269.png, https://aws1.discourse-cdn.com/github/optimized/1X/9cd67d9bf8f8127b733108cdb467d1562c3e2002_2_1033x403.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/9cd67d9bf8f8127b733108cdb467d1562c3e2002.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/9cd67d9bf8f8127b733108cdb467d1562c3e2002_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">gh-ss2.png</span><span class="informations">1195×467 21.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/b72f6d3c976e3a1dd9f53a52cb896bfc05834d31.png" data-download-href="/uploads/short-url/q8wZhbgLOE60BIiqFSoMe0AleZr.png?dl=1" title="gh-ss1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/20893901/181067468-20cd357b-cbd7-4994-8e69-a0e976e0b130.png" alt="gh-ss1.png" data-base62-sha1="q8wZhbgLOE60BIiqFSoMe0AleZr" width="690" height="260" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/b72f6d3c976e3a1dd9f53a52cb896bfc05834d31_2_690x260.png, https://aws1.discourse-cdn.com/github/optimized/1X/b72f6d3c976e3a1dd9f53a52cb896bfc05834d31_2_1035x390.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/b72f6d3c976e3a1dd9f53a52cb896bfc05834d31.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/b72f6d3c976e3a1dd9f53a52cb896bfc05834d31_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">gh-ss1.png</span><span class="informations">1210×456 21.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by birdcar** [comment, marked as answer]
<p>Hi <a class="mention" href="/yotzu">@yotzu</a>! </p>
<p>I took a look at the repositories associated with your GitHub user and I think I figured out what’s going on here. </p>
<p>Domain inheritance on GitHub Pages, where your custom domain automatically applies to Project Pages sites, only occurs when your custom domain is configured with a User Pages site. </p>
<p><a href="https://help.github.com/articles/user-organization-and-project-pages/" rel="nofollow noopener">https://help.github.com/articles/user-organization-and-project-pages/</a></p>
<p>Because your custom domain, <a href="http://tidyrabbit.com" rel="nofollow noopener">tidyrabbit.com</a>, is configured with your “home” repository, it’s not being applied to every other Pages site you have. If you’d like to change this, you can rename your “home” repository to be named “<a href="http://yotzu.github.io" rel="nofollow noopener">yotzu.github.io</a>” and your custom domain will then be used by your “test” repository the way you expect it to be.</p>

**Comment by yotzu** [comment]
<p>Yes, that was exactly the problem! Thank you so very much! This has been bugging me for days now</p>
